streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,16 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners", classOf[BatchInfoCollector].getName + "," +
```

Comment:
Also please include a third case where the incorrect StreamingListener with 2 args is specified, which is expected throw an exception. Test with `intercept[SparkException]` and check whether the exception message has "Exception when registering Streaming Listener:" and the class name".